test/sequential/test-async-wrap-getasyncid.js

```diff
@@ -247,7 +247,24 @@ if (common.hasCrypto) { // eslint-disable-line crypto-check
 
 {
   // Do our best to grab a tty fd.
-  const tty_fd = common.getTTYfd();
+  function getTTYfd() {
+    const tty = require('tty');
+    let tty_fd = 0;
+    if (!tty.isatty(tty_fd)) tty_fd++;
+    else if (!tty.isatty(tty_fd)) tty_fd++;
+    else if (!tty.isatty(tty_fd)) tty_fd++;
```

Comment:
Can someone explain to me how these three lines work in the first place?

Reply:
I didn't write the code but I assume:

The default file descriptors for I/O are stdin, stdout and stderr (0, 1 and 2 respectively). It's not _always_ guaranteed to be the case, so we try the first 3 and then try `/dev/tty` which usually works (it maps for the controlling terminal of a process for each process) but it's slower to call than isatty it if it's 0.

Reply:
It was a cut-and-paste, but yeah, it's not just hard to follow, but also buggy. If the first `if` fails, on line 253, it increments `tty_fd` but doesn't then move to line 254 because that's an `else` so it skips that land. Ugh.

If 0 and 1 are not TTYs but 2 is, we won't get 2 (unless that's what's returned by the `try/catch` block subsequently.

I think something more like this is what we want:

```js
let tty_fd = [0, 1, 2].find(() => tty.isatty)
if (tty_fd === undefined) {
         try {
        tty_fd = fs.openSync('/dev/tty');
      } catch (e) {
        // There aren't any tty fd's available to use.
        return -1;
      }
return tty_fd;
```

(And while I'm in there fixing this function so it hopefully works correctly, I should probably refactor `tty_fd` to have a name that doesn't include `_`.)

Since this PR is a simple copy/paste code move with several approvals that doesn't introduce any bug  or hard to read code that isn't already in our code base, I'd prefer to land it and save the bug-fix and refactoring for a follow-up PR.

Reply:
Added a commit here to fix this. PTAL.